pkg/controller/controller_utils.go

```diff
@@ -133,10 +133,10 @@ type ControllerExpectations struct {
 
 // GetExpectations returns the ControlleeExpectations of the given controller.
 func (r *ControllerExpectations) GetExpectations(controllerKey string) (*ControlleeExpectations, bool, error) {
-	if exp, exists, err := r.GetByKey(controllerKey); err == nil && exists {
+	if exp, exists, _ := r.GetByKey(controllerKey); exists {
```

Comment:
As long as the function signature includes an error, we need to continue checking it regardless of what the current implementation does.  As I recall, that method is on an interface, so other/future implementations may return an error.